### PR TITLE
fix(module: date-picker): fix year-picker and month-picker style erro…

### PR DESCRIPTION
**Pull request by hungtcs**
…r within compacted input group

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [2084](https://github.com/NG-ZORRO/ng-zorro-antd/issues/2084)

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/date-picker/month-picker.component.spec.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ describe('NzMonthPickerComponent', () => {
     it('should support nzClassName', () => {
       const className = fixtureInstance.nzClassName = 'my-test-class';
       fixture.detectChanges();
-      const picker = debugElement.query(By.css('.ant-calendar-picker')).nativeElement as HTMLElement;
+      const picker = debugElement.queryAll(By.css('.ant-calendar-picker'))[1].nativeElement as HTMLElement;
```
Comment (vthinkxie) [inline components/date-picker/month-picker.component.spec.ts]: it seems it has nothing todo with the pr with this code.

Comment (hungtcs) [reply components/date-picker/month-picker.component.spec.ts]: @vthinkxie It's hard to test styles via javascript, see this [stackblitz](https://stackblitz.com/edit/ng-zorro-antd-start-yp2rts)

Comment (vthinkxie) [reply components/date-picker/month-picker.component.spec.ts]: @hungtcs the test is must for ci, you can use https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle to test the style in javascript.

Comment (hungtcs) [reply components/date-picker/month-picker.component.spec.ts]: @vthinkxie Thank you, I will add the test case later.

Comment (vthinkxie) [reply components/date-picker/month-picker.component.spec.ts]: 👍

Comment (hungtcs) [reply components/date-picker/month-picker.component.spec.ts]: @vthinkxie Test case is completed.

**Pull request merged**